flink/v2.1/flink/src/main/java/org/apache/iceberg/flink/TypeToFlinkType.java

```diff
@@ -83,6 +84,11 @@ public LogicalType map(Types.MapType map, LogicalType keyResult, LogicalType val
     return new MapType(keyResult.copy(false), valueResult.copy(map.isValueOptional()));
   }
 
+  @Override
+  public LogicalType variant(Types.VariantType variant) {
+    return new VariantType(false); // VariantType is always non-nullable in Flink
```

Comment:
> VariantType is always non-nullable in Flink

Is this correct? I saw plenty of places in Flink where nullable variant type is used.

https://github.com/search?q=repo%3Aapache%2Fflink%20%22new%20VariantType()%22&type=code